web/src/pm_list.ts

```diff
@@ -144,6 +144,12 @@ function scroll_all_private_into_view(): void {
 
 export function handle_narrow_activated(filter: Filter): void {
     const active_filter = filter;
+    const is_private_messages_in_view = active_filter.has_operator("dm");
+
+    if (!is_private_messages_in_view) {
+        update_private_messages();
+        return;
+    }
```

Comment:
This I think needs to be further down to avoid short-circuiting the `is_all_private_message_view` logic.

Reply:
Also, why is `update_private_messages` the right thing to call in this case?

Reply:
> Also, why is `update_private_messages` the right thing to call in this case?

It isn't the only thing that can be called, we could do something like `deselect_dm_row` and not build the dm list again just like in `left_sidebar_navigation_area.handle_narrow_activated()`.

Reply:
I guess the DM will potentially still be visible, but it is probably enough of a waste of resources that we should try to avoid it.